src/data_types/link_secret.rs

```diff
@@ -1,16 +1,15 @@
-use crate::cl::{bn::BigNumber, LinkSecret as ClLinkSecret, Prover as CryptoProver};
-use crate::error::ConversionError;
 use std::fmt;
 
-pub struct LinkSecret(pub BigNumber);
+use crate::cl::{bn::BigNumber, Prover as CryptoProver};
+use crate::error::ConversionError;
+
+pub struct LinkSecret(pub(crate) BigNumber);
```

Comment:
Would it be possible to derive `zeroize` on this? It was not done before due to Ursa but it might be possible now.

Reply:
For version 0.2.2 the crate is using openssl's secure context for big numbers, which as I understand it zeroizes the entire buffer before it's released. So this type could implement ZeroizeOnDrop, but probably not Zeroize.